Exercise_3.py

```diff
@@ -3,6 +3,8 @@ class ListNode:
     A node in a singly-linked list.
     """
     def __init__(self, data=None, next=None):
+        self.data = data
+        self.next = None
```

Comment:
Please review  the `Sample.java` file and include Time and Space Complexity for all program files. Also, specify if the code is executed as expected or not.